src/ServiceStatus.ts

```diff
@@ -58,13 +58,39 @@ export class ServiceStatus {
         this.observers.forEach(observer => observer.updateState(this.State, this.response));
     }
 
+    private hasDeploymentHeaders(expected: Response, response: Response) {
+        let res: boolean[] = [];
+        expected.headers.forEach((value, key) => {
+            if (response.headers.has(key)) {
+                res.push(true);
+            }
+        });
+
+        if (res.length) {
+            return res.every(Boolean);
+        }
+        return false;
+    }
+
+    private compare(response: Response, expected: Response): boolean {
+        let result = false;
+        expected.status && (result = expected.status === response.status);
+        return result;
+    }
+
     private async callback() {
         if (!this.ping) return;
         await this.ping.ping()
             .then(async (response: Response) => {
                 this.response = response;
-                if (this.response.status === 200) return this.goOnline();
-                else return this.goOffline();        
+                if (this.compare(this.response, this.expected)) {
+                    if (this.hasDeploymentHeaders(this.expected, this.response)) {
```

Comment:
no additional checks here. put this inside compare or avoid it completelty. leaky abstraction.